lib/msf/util/dot_net_deserialization/types/primitives.rb

```diff
@@ -0,0 +1,203 @@
+module Msf
+module Util
+module DotNetDeserialization
+module Types
+module Primitives
+
+  #
+  # .NET Serialization Types (Primitives)
+  #
+  class Boolean < BinData::Primitive
+    int8 :val
+    def get
+      self.val != 0
+    end
+
+    def set(value)
+      self.val = value ? 1 : 0
+    end
+  end
+
+  class DateTime < BinData::Primitive
+    # see: https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-nrbf/f05212bd-57f4-4c4b-9d98-b84c7c658054
+    default_parameter kind_name: :unspecified
+
+    endian   :little
+    sbit62   :ticks
+    bit2     :kind, initial_value: -> { kind_initial_value }
+
+    KindEnum = {
+        unspecified: 0,
+        utc:         1,
+        local:       2
+    }
```

Comment:
Minor consistency nitpick here, but do we use `name: value` or `:name => value` when defining enums? We have a mix at the mo :)